apps/mason2/vcf_materialization.cpp

```diff
@@ -214,6 +214,9 @@ void VcfMaterializer::init()
                 throw MasonIOException("Could not write methylation levels FAI index.");
         }
     }
+
+    if (!empty(vcfFileName) && numSeqs(faiIndex) != length(contigNames(context(vcfFileIn))))
+        throw MasonIOException("Number of records in FAI index and VCF differ.");
```

Comment:
"records" is a bad name. Would `contigs` be better?

Reply:
Does the error persist if you remove this dependency again? (With only the fix below?)

I still think from a semantic POV that they may differ.

I think it is valid to only apply mutations to a specific chromosome while still simulating from the entire reference sequence.

Reply:
Semantically, they might differ. But I have no idea where to adapt the code for that.
The code definitely expects to have the same number of contigs in FASTA and VCF; you'll go out of bounds if they are not the same.

Reply:
ah I see, then we should add this as a quick fix :+1: